src/etl/extract.py

```diff
@@ -0,0 +1,85 @@
+import polars as pl
+from glob import glob
+from os import path
+from logging import Logger
+
+current_file_dir = path.dirname(path.realpath(__file__))
+
+
+class Extract:
+    def __init__(self, config: dict, logger: Logger) -> None:
```

Comment:
If you wanted to 'gold plate' this solution, you could use a `TypedDict` for config, with valid keys.

https://docs.python.org/3/library/typing.html#typing.TypedDict